### ISSUE TITLE
OBF files missing many foot designation and access restriction tags

### ISSUE THREAD

**Issue by Wonderer0** [osmandapp/OsmAnd-tools#654]
I am trying to write a custom `render.xml` that colours bridleways, footways, paths, and tracks according to the right to access them by foot.  The colour is specified by a `<renderingAttribute>` that tests the various `designation=*` and `foot=*` tags to return the correct `attrColorValue`.

Many of them default to grey because none of the tests pass, even though the corresponding Ways given on the OSM website have at least one qualifying tag/value pair that should be caught by the tests.  The only explanation seemed to be that some tags were missing from the OBF file.

To check this I:

- Downloaded and unpacked [Gb_england_south-east-england_europe_2.obf.zip](https://osmand.net/download?standard=yes&file=Gb_england_south-east-england_europe_2.obf.zip). 

- Downloaded and installed [OsmAndMapCreator](https://wiki.openstreetmap.org/wiki/OsmAndMapCreator).

- Generated a CSV file by running:
`inspector -vmap -vmapobjects -v "<download dir>\Gb_england_south-east-england_europe_2.obf" >obf.txt`

The OBF source file format is completely new to me, but seems to confirm that designations and access restrictions given on OSM are lost, sometimes several of them from the same Way, and not just footpath designations.  Below are 4 examples that give the tags shown on OSM and the corresponding lines from obf.csv:

- Footpath [#93141830](https://www.openstreetmap.org/way/93141830) [designation=public_footpath, foot=designated, highway=footway, horse=no, vehicle=no]
Has lost both ways of designating it as a public footpath, but preserved no-access for both horse and vehicle:
`Way types [highway-footway (5)] add_types [vehicle-no(305), horse-no(84)] id 11922154271 osmid 93141830`

- Bridleway [#31650970](https://www.openstreetmap.org/way/31650970) [bicycle=designated, designation=public_bridleway, foot=designated, highway=bridleway, horse=designated, motor_vehicle=no]
Has lost `designation=public_bridleway` , ` foot=designated` and `horse=designated`, but again has preserved bicycle designation and vehicle no-access:
`Way types [highway-bridleway (112)] add_types [motor_vehicle-no(121), bicycle-designated(97), osmand_highway_integrity-0(4)] id 4051324243 osmid 31650970`
- Path [#629989520](https://www.openstreetmap.org/way/629989520)  [designation=public_footpath, highway=path]
Has lost its footpath designation:
`Way types [highway-path (25)] id 80638658651 osmid 629989520`

- Track [#904063460](https://www.openstreetmap.org/way/904063460) [bicycle=designated, designation=byway_open_to_all_traffic, foot=designated, highway=track, horse=designated, motor_vehicle=yes, vehicle=yes]
Has lost both the designations that give foot access (designation=byway_open_to_all_traffic, foot=designated), but preserved the 3 that give access to bikes and motor vehicles:
`Way types [highway-track (24)] add_types [vehicle-yes(501), motor_vehicle-yes(328), bicycle-designated(97)] id 115720122919 osmid 904063460`

This is a very small sample of the many Ways that have lost information, but is there a pattern here in that information about foot access (human and horse) is lost but about wheel access (bicycle and vehicle) is preserved?

Footpath designations and access restrictions are arguably the most important tags for walkers, especially in the UK countryside with its patchwork of various types of paths with different restrictions.  I had hoped to able to use OsmAnd with the custom renderer to be able see at a glance how various local paths are tagged in OSM.  This information would not only help when planning walks, but also make it easy to compare what's on the ground against what's in the OSM database, and correct it there when there are anomalies.   This depends upon the OBF files fully and accurately reflecting the tags on OSM.

I am posting the issue here because I assume the missing data is due to bug(s) in OsmAndMapCreator. 

Is there any way round the problem until the bugs are fixed?

**Comment by vshcherb** [comment]
It's controlled by https://github.com/osmandapp/OsmAnd-resources/blob/master/obf_creation/rendering_types.xml. We don't any data unless we plan to render them, binary format is specific for rendering purposes so it's simplified.

if you run inspector -vrouting - you will see different tags which are closer to original and these tags are used for routing.

**Comment by Wonderer0** [comment]
Thanks for the link to the `rendering_types.xml` file.

I produced a modified version of it that adds `designation` and extra `foot` tags, and also omits some unwanted transformations.  I was then able to use `OsmAndMapCreator/utilities.bat` to create a custom OBF file for South-East England from a PBF file at [download.openstreetmap.fr](http://download.openstreetmap.fr/extracts/europe/united_kingdom/england/).  It produced 244 errors of the form:
```
! Error parsing float number ... SW to its jn w/ Headington Road
! Parsed number -> '0.0 SW to its jn w/ Headington Road'
```
189 were that error exactly.

Is this normal?  It produced an OBF file that seems to be OK, but has important information been lost?

P.S.
I have produced a custom `render.xml` that colours ways according to their access rights for walkers, and also shows where public footpaths and bridleways go along tracks and roads.  As this might also be helpful for other walkers, especially in the UK, I would be happy to publish it.  Can I do this through one of the [osmandapp](https://github.com/osmandapp) repositories?  Any advice would be welcome.  Should I raise this as a separate discussion?
